lib/workers/repository/update/pr/changelog/bitbucket/index.spec.ts

```diff
@@ -0,0 +1,158 @@
+import { ChangeLogProject, ChangeLogRelease, getChangeLogJSON } from '..';
+import { Fixtures } from '../../../../../../../test/fixtures';
+import * as httpMock from '../../../../../../../test/http-mock';
+import { partial } from '../../../../../../../test/util';
+import * as semverVersioning from '../../../../../../modules/versioning/semver';
+import type { BranchUpgradeConfig } from '../../../../../types';
+import { getReleaseNotesMd } from '../release-notes';
+
+jest.mock('../../../../../../modules/datasource/npm');
+
+const changelogMd = Fixtures.get('gitter-webapp.md', '../..');
+
+const upgrade = partial<BranchUpgradeConfig>({
+  manager: 'some-manager',
+  branchName: '',
+  endpoint: 'https://api.bitbucket.org/',
+  packageName: 'some-repo',
+  versioning: semverVersioning.id,
+  currentVersion: '5.2.0',
+  newVersion: '5.7.0',
+  sourceUrl: 'https://bitbucket.org/some-org/some-repo',
+  releases: [
+    { version: '5.2.0' },
+    {
+      version: '5.4.0',
+      releaseTimestamp: '2018-08-24T14:23:00.000Z',
+    },
+    { version: '5.5.0', gitRef: 'abcd' },
+    {
+      version: '5.6.0',
+      gitRef: '1234',
+      releaseTimestamp: '2020-02-13T15:37:00.000Z',
+    },
+    { version: '5.6.1', gitRef: 'asdf' },
+  ],
+});
+
+const bitbucketTreeResponse = {
+  values: [
+    {
+      type: 'commit_directory',
+      path: 'lib',
+      commit: {
+        hash: '1234',
+      },
+    },
+    {
+      type: 'commit_file',
+      path: 'CHANGELOG.md',
+      commit: {
+        hash: 'abcd',
+      },
+    },
+    {
+      type: 'commit_file',
+      path: 'RELEASE_NOTES.md',
+      commit: {
+        hash: 'asdf',
+      },
+    },
+  ],
+};
+
+const bitbucketTreeResponseNoChangelogFiles = {
+  values: [
+    {
+      type: 'commit_directory',
+      path: 'lib',
+      commit: {
+        hash: '1234',
+      },
+    },
+  ],
+};
+
+const bitbucketProject = partial<ChangeLogProject>({
+  type: 'bitbucket',
+  apiBaseUrl: 'https://api.bitbucket.org',
+  baseUrl: 'https://bitbucket.org/',
+});
+
+describe('workers/repository/update/pr/changelog/bitbucket/index', () => {
+  afterEach(() => {
+    // FIXME: add missing http mocks
+    httpMock.clear(false);
```

Comment:
please fix

Reply:
Thanks for all the feedback.  I ended up starting the test again and getting rid of all the legacy baggage i was incorrectly trying to carry